ethcore/src/externalities.rs

```diff
@@ -411,7 +412,25 @@ impl<'a, T: 'a, V: 'a, B: 'a> Ext for Externalities<'a, T, V, B>
 	}
 
 	fn trace_executed(&mut self, gas_used: U256, stack_push: &[U256], mem_diff: Option<(usize, &[u8])>, store_diff: Option<(U256, U256)>) {
-		self.vm_tracer.trace_executed(gas_used, stack_push, mem_diff, store_diff)
+		let Externalities { ref state, ref mut vm_tracer, .. } = *self;
+
+		let access = ExtStorageAccess(*state, &self.origin_info.address);
```

Comment:
Would it be possible to change to a design where if `NoopTracer` is used, we don't create this struct? For example, this can be done by directly passing `State` reference into tracer.

Reply:
The benefit is pairing the storage access with the address of the contract so that we don't have to do this, making the storage access API more convenient.

What are the worries about from setting up this struct? The overhead should be negligible since we are mostly dealing with pointers.

Reply:
Yeah looks like the overhead is negligible. However, I wonder about other tracer usage for ethcore-as-a-library. People may want to build other tracer implementation that access to other state information (like other account's storage slot, account code, nonce, etc). So I'm just thinking maybe directly passing `State` would be more generalized and potentially fit more usage?